components/builder-sessionsrv/src/server/mod.rs

```diff
@@ -64,30 +66,51 @@ lazy_static! {
     };
 }
 
-#[derive(Clone)]
+pub type SessionKey = (u32, proto::OAuthProvider);
+
+#[derive(Clone, Debug)]
 pub struct Session {
     pub created_at: Instant,
+    pub key: SessionKey,
     inner: proto::Session,
+    token: proto::SessionToken,
 }
 
 impl Session {
-    pub fn expired(&self) -> bool {
-        self.created_at.elapsed() >= *SESSION_DURATION
+    pub fn build(
+        mut msg: proto::SessionCreate,
+        mut account: proto::Account,
+        flags: FeatureFlags,
+    ) -> SrvResult<Self> {
+        let mut session = proto::Session::new();
+        let mut token = proto::SessionToken::new();
+        token.set_account_id(account.get_id());
+        token.set_extern_id(msg.get_extern_id());
+        token.set_provider(msg.get_provider());
+        token.set_token(msg.get_token().to_string().into_bytes());
+
+        session.set_id(account.get_id());
```

Comment:
rename session "id" field to "account_id"?

Reply:
Yeah, agreed

Reply:
Although that will break a ton of other code that all references `session.get_id()`.  😞

Reply:
On second thought, maybe it's worth leaving it so we're not changing too many things at once.

Reply:
Yeah we can take a pass at all of the message names/IDs if we'd like just before launch. It's all typed so it should be an easy enough refactor. We have dead messages, fields that no longer are needed, and at this point we don't need to be backwards compatible because we're going to take an outage when we deploy to prod anyway